flowmachine/flowmachine/features/subscriber/contact_balance.py

```diff
@@ -138,21 +144,17 @@ def _make_query(self):
             SELECT
                 subscriber,
                 count(*) AS events
-            FROM (
-              SELECT DISTINCT U.subscriber,
```

Comment:
Just to check my understanding: how does removing the `DISTINCT` here (and below) change the logic what's in the result?

Reply:
The `DISTINCT` was here to ensure we did not double count events. We used distinct on subscriber, counterpart and id, since we only want to count this triplet once. Since we removed the id from this query, if we use `DISTINCT` we will basically select distinct pairs of subscriber and counterparts which will reduce all the events between both to a single line precluding us from counting the events which we want to count in the first place.